pkg/functions/runner.go

```diff
@@ -153,31 +149,27 @@ func runGo(ctx context.Context, job *Job) (err error) {
 }
 
 func waitFor(job *Job, timeout time.Duration) error {
-	var (
-		url  = fmt.Sprintf("http://%s:%s/%s", job.Host, job.Port, readinessEndpoint)
-		tick = time.NewTicker(200 * time.Millisecond)
-	)
-	defer tick.Stop()
+	url := fmt.Sprintf("http://%s:%s/%s", job.Host, job.Port, readinessEndpoint)
 	if job.verbose {
 		fmt.Printf("Waiting for %v\n", url)
 	}
 	for {
 		select {
 		case <-time.After(timeout):
 			return errors.New("timed out waiting for function to be ready")
-		case <-tick.C:
+		case <-time.After(200 * time.Millisecond):
 			resp, err := http.Get(url)
 			defer resp.Body.Close()
```

Comment:
This may lead to `Close()` on `nil` body.

Reply:
Also `defer` in a loop is not good idea.

Reply:
Yes that seems to be the problem - I confirmed by moving defer adter error check and make test work https://github.com/knative/func/issues/1801

Reply:
#1801 is now merged, thanks @aslom 
#1804 is now open which incorporates your suggestion here @matej and Lance's suggestion to expand the typed timeout error.